expression/builtin_time.go

```diff
@@ -5629,3 +5630,50 @@ func getExpressionFsp(ctx sessionctx.Context, expression Expression) (int, error
 	}
 	return mathutil.Min(expression.GetType().Decimal, types.MaxFsp), nil
 }
+
+type tsoFunctionClass struct {
+	baseFunctionClass
+}
+
+func (c *tsoFunctionClass) getFunction(ctx sessionctx.Context, args []Expression) (builtinFunc, error) {
+	if err := c.verifyArgs(args); err != nil {
+		return nil, errors.Trace(err)
+	}
+	//bf := newBaseBuiltinFuncWithTp(ctx, args, types.ETDatetime, types.ETDatetime)
+	argTp := args[0].GetType().EvalType()
+	bf := newBaseBuiltinFuncWithTp(ctx, args, argTp, types.ETInt)
+
+	bf.tp.Tp, bf.tp.Flen, bf.tp.Decimal = mysql.TypeDate, mysql.MaxDateWidth, types.DefaultFsp
+	sig := &builtinTsoSig{bf}
+	return sig, nil
+}
+
+type builtinTsoSig struct {
+	baseBuiltinFunc
+}
+
+func (b *builtinTsoSig) Clone() builtinFunc {
+	newSig := &builtinTsoSig{}
+	newSig.cloneFrom(&b.baseBuiltinFunc)
+	return newSig
+}
+
+// evalTime evals a builtinTsoSig.
+func (b *builtinTsoSig) evalTime(row chunk.Row) (types.Time, bool, error) {
+	arg, isNull, err := b.args[0].EvalInt(b.ctx, row)
+	if isNull || err != nil || arg <= 0 {
```

Comment:
Why do we consider the case of ``isNull == true`` as an error ?

Reply:
function must accept a parameter

Reply:
got it.